extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/user/UserService.java

```diff
@@ -88,6 +88,7 @@ private void putAllUsers(Map<String, User> users, LDAPConnection ldapConnection,
             // Set search limits
             LDAPSearchConstraints constraints = new LDAPSearchConstraints();
             constraints.setMaxResults(confService.getMaxResults());
+            constraints.setDereference(confService.getDereferenceAliases());
```

Comment:
If we keep building a new `LDAPSearchConstraints` object, would it make more sense to provide a function which does so within `ConfigurationService`, rather than repeatedly calling the same setters on a new `LDAPSearchConstraints` prior to each LDAP query?

I'm also not clear on why `setMaxResults()` is only applied here. If it should apply universally, then I would lean even more strongly in the direction described above. We can abstract away the configuration properties which apply to LDAP constraints within the internals of `ConfigurationService` and just rely on the service to return an appropriate constraints object.